target/scripts/helper-functions.sh

```diff
@@ -167,7 +167,7 @@ for key, value in acme.items():
 
   local CERT
   # shellcheck disable=SC2002
-  CERT=$(cat /etc/letsencrypt/acme.json | python -c "
+  CERT=$(cat /etc/letsencrypt/acme.json | python3 -c "
```

Comment:
Do we require python in our image only to do that?

Reply:
No, another dependency [`postfix-policyd-spf-python`](https://packages.debian.org/bullseye/postfix-policyd-spf-python) depends on it, I'm not sure if anything else does:

<details>
<summary>revere dependencies on bullseye upgrade for python3</summary>

```console
$ apt-cache rdepends --installed python3  
python3
Reverse Depends:
  postfix
```

```console
$ apt-rdepends python3
python3
  Depends: libpython3-stdlib (= 3.9.2-3)
  Depends: python3.9 (>= 3.9.2-0~)
  PreDepends: python3-minimal (= 3.9.2-3)
libpython3-stdlib
  Depends: libpython3.9-stdlib (>= 3.9.2-0~)
libpython3.9-stdlib
  Depends: libbz2-1.0
  Depends: libc6 (>= 2.28)
  Depends: libcrypt1 (>= 1:4.1.0)
  Depends: libdb5.3
  Depends: libffi7 (>= 3.3~20180313)
  Depends: liblzma5 (>= 5.1.1alpha+20120614)
  Depends: libmpdec3
  Depends: libncursesw6 (>= 6)
  Depends: libnsl2 (>= 1.0)
  Depends: libpython3.9-minimal (= 3.9.2-1)
  Depends: libreadline8 (>= 7.0~beta)
  Depends: libsqlite3-0 (>= 3.7.15)
  Depends: libtinfo6 (>= 6)
  Depends: libtirpc3 (>= 1.0.2)
  Depends: libuuid1 (>= 2.20.1)
  Depends: media-types
  Depends: mime-support
  Depends: tzdata
libbz2-1.0
  Depends: libc6 (>= 2.4)
libc6
  Depends: libcrypt1
  Depends: libgcc-s1
libcrypt1
  Depends: libc6 (>= 2.25)
libgcc-s1
  Depends: gcc-10-base (= 10.2.1-6)
  Depends: libc6 (>= 2.14)
gcc-10-base
libdb5.3
  Depends: libc6 (>= 2.17)
libffi7
  Depends: libc6 (>= 2.14)
liblzma5
  Depends: libc6 (>= 2.17)
libmpdec3
  Depends: libc6 (>= 2.14)
  Depends: libgcc-s1 (>= 3.0)
  Depends: libstdc++6 (>= 5.2)
libstdc++6
  Depends: gcc-10-base (= 10.2.1-6)
  Depends: libc6 (>= 2.23)
  Depends: libgcc-s1 (>= 4.2)
libncursesw6
  Depends: libc6 (>= 2.14)
  Depends: libtinfo6 (= 6.2+20201114-2)
libtinfo6
  Depends: libc6 (>= 2.16)
libnsl2
  Depends: libc6 (>= 2.14)
  Depends: libtirpc3 (>= 1.0.2)
libtirpc3
  Depends: libc6 (>= 2.16)
  Depends: libgssapi-krb5-2 (>= 1.17)
  Depends: libtirpc-common (>= 1.3.1-1)
libgssapi-krb5-2
  Depends: libc6 (>= 2.27)
  Depends: libcom-err2 (>= 1.43.9)
  Depends: libk5crypto3 (>= 1.16)
  Depends: libkrb5-3 (= 1.18.3-6+deb11u1)
  Depends: libkrb5support0 (>= 1.15~beta1)
libcom-err2
  Depends: libc6 (>= 2.17)
libk5crypto3
  Depends: libc6 (>= 2.25)
  Depends: libkrb5support0 (>= 1.16)
libkrb5support0
  Depends: libc6 (>= 2.25)
libkrb5-3
  Depends: libc6 (>= 2.25)
  Depends: libcom-err2 (>= 1.43.9)
  Depends: libk5crypto3 (>= 1.15~beta1)
  Depends: libkeyutils1 (>= 1.5.9)
  Depends: libkrb5support0 (= 1.18.3-6+deb11u1)
  Depends: libssl1.1 (>= 1.1.0)
libkeyutils1
  Depends: libc6 (>= 2.14)
libssl1.1
  Depends: debconf (>= 0.5)
  Depends: debconf-2.0
  Depends: libc6 (>= 2.25)
debconf
  PreDepends: perl-base (>= 5.20.1-3~)
perl-base
  PreDepends: dpkg (>= 1.17.17)
  PreDepends: libc6 (>= 2.29)
  PreDepends: libcrypt1 (>= 1:4.1.0)
dpkg
  Depends: tar (>= 1.28-1)
  PreDepends: libbz2-1.0
  PreDepends: libc6 (>= 2.15)
  PreDepends: liblzma5 (>= 5.2.2)
  PreDepends: libselinux1 (>= 3.1~)
  PreDepends: zlib1g (>= 1:1.1.4)
tar
  PreDepends: libacl1 (>= 2.2.23)
  PreDepends: libc6 (>= 2.28)
  PreDepends: libselinux1 (>= 3.1~)
libacl1
  Depends: libc6 (>= 2.14)
libselinux1
  Depends: libc6 (>= 2.14)
  Depends: libpcre2-8-0 (>= 10.22)
libpcre2-8-0
  Depends: libc6 (>= 2.14)
zlib1g
  Depends: libc6 (>= 2.14)
debconf-2.0
libtirpc-common
libpython3.9-minimal
  Depends: libc6 (>= 2.4)
  Depends: libssl1.1 (>= 1.1.1)
libreadline8
  Depends: libc6 (>= 2.15)
  Depends: libtinfo6 (>= 6)
  Depends: readline-common
readline-common
  Depends: dpkg (>= 1.15.4)
  Depends: install-info
install-info
  Depends: libc6 (>= 2.14)
  PreDepends: dpkg (>= 1.16.1)
libsqlite3-0
  Depends: libc6 (>= 2.29)
libuuid1
  Depends: libc6 (>= 2.25)
media-types
mime-support
  Depends: mailcap
  Depends: media-types
mailcap
  Depends: media-types
  Depends: perl
perl
  Depends: libperl5.32 (= 5.32.1-4+deb11u2)
  Depends: perl-base (= 5.32.1-4+deb11u2)
  Depends: perl-modules-5.32 (>= 5.32.1-4+deb11u2)
  PreDepends: dpkg (>= 1.17.17)
libperl5.32
  Depends: libbz2-1.0
  Depends: libc6 (>= 2.29)
  Depends: libcrypt1 (>= 1:4.1.0)
  Depends: libdb5.3
  Depends: libgdbm-compat4 (>= 1.18-3)
  Depends: libgdbm6 (>= 1.18-3)
  Depends: perl-modules-5.32 (>= 5.32.1-4+deb11u2)
  Depends: zlib1g (>= 1:1.2.2.3)
libgdbm-compat4
  Depends: libc6 (>= 2.14)
  Depends: libgdbm6 (>= 1.16)
libgdbm6
  Depends: libc6 (>= 2.14)
perl-modules-5.32
  Depends: perl-base (>= 5.32.1-1)
  PreDepends: dpkg (>= 1.17.17)
tzdata
  Depends: debconf (>= 0.5)
  Depends: debconf-2.0
python3.9
  Depends: libpython3.9-stdlib (= 3.9.2-1)
  Depends: media-types
  Depends: mime-support
  Depends: python3.9-minimal (= 3.9.2-1)
python3.9-minimal
  Depends: libexpat1 (>= 2.1~beta3)
  Depends: libpython3.9-minimal (= 3.9.2-1)
  Depends: zlib1g (>= 1:1.2.0)
  PreDepends: libc6 (>= 2.29)
libexpat1
  Depends: libc6 (>= 2.25)
python3-minimal
  Depends: dpkg (>= 1.13.20)
  PreDepends: python3.9-minimal (>= 3.9.2-0~)
```

```console
$ apt-rdepends python3.9-minimal
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python3.9-minimal
  Depends: libexpat1 (>= 2.1~beta3)
  Depends: libpython3.9-minimal (= 3.9.2-1)
  Depends: zlib1g (>= 1:1.2.0)
  PreDepends: libc6 (>= 2.29)
libexpat1
  Depends: libc6 (>= 2.25)
libc6
  Depends: libcrypt1
  Depends: libgcc-s1
libcrypt1
  Depends: libc6 (>= 2.25)
libgcc-s1
  Depends: gcc-10-base (= 10.2.1-6)
  Depends: libc6 (>= 2.14)
gcc-10-base
libpython3.9-minimal
  Depends: libc6 (>= 2.4)
  Depends: libssl1.1 (>= 1.1.1)
libssl1.1
  Depends: debconf (>= 0.5)
  Depends: debconf-2.0
  Depends: libc6 (>= 2.25)
debconf
  PreDepends: perl-base (>= 5.20.1-3~)
perl-base
  PreDepends: dpkg (>= 1.17.17)
  PreDepends: libc6 (>= 2.29)
  PreDepends: libcrypt1 (>= 1:4.1.0)
dpkg
  Depends: tar (>= 1.28-1)
  PreDepends: libbz2-1.0
  PreDepends: libc6 (>= 2.15)
  PreDepends: liblzma5 (>= 5.2.2)
  PreDepends: libselinux1 (>= 3.1~)
  PreDepends: zlib1g (>= 1:1.1.4)
tar
  PreDepends: libacl1 (>= 2.2.23)
  PreDepends: libc6 (>= 2.28)
  PreDepends: libselinux1 (>= 3.1~)
libacl1
  Depends: libc6 (>= 2.14)
libselinux1
  Depends: libc6 (>= 2.14)
  Depends: libpcre2-8-0 (>= 10.22)
libpcre2-8-0
  Depends: libc6 (>= 2.14)
libbz2-1.0
  Depends: libc6 (>= 2.4)
liblzma5
  Depends: libc6 (>= 2.17)
zlib1g
  Depends: libc6 (>= 2.14)
debconf-2.0
```

</details>

I doubt removing python will save much image size ratio-wise if that's what you were thinking. Your welcome to replace the `acme.json` parsing/extraction script and the SPF policy server for Postfix, but I doubt it's worth the effort :sweat_smile: 

I've sussed out the test failure (_the python code for acme.json is presently Python 2 apparently, so it needed to be updated to v3_).

Reply:
I don't think removing python is a good idea too. We should leave it in there, but we should also use Python 3 :D